applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_adv.c

```diff
@@ -0,0 +1,232 @@
+		adv_param.options |= BT_LE_ADV_OPT_DIR_ADDR_RPA;
+
+		/* Clear ADV data set before update to direct advertising */
+		ret = bt_le_ext_adv_set_data(adv_ext, NULL, 0, NULL, 0);
+		if (ret) {
+			LOG_ERR("Failed to clear advertising data. Err: %d", ret);
+			return;
+		}
+
+		ret = bt_le_ext_adv_update_param(adv_ext, &adv_param);
+		if (ret) {
+			LOG_ERR("Failed to update ext_adv to direct advertising. Err = %d", ret);
+			return;
+		}
+
+		bt_addr_le_to_str(&addr, addr_buf, BT_ADDR_LE_STR_LEN);
+		LOG_INF("Set direct advertising to %s", addr_buf);
+	} else
+#endif /* CONFIG_BT_BONDABLE */
+	{
+		if (ad_peer_local == NULL) {
+			LOG_ERR("Ad_peer not set");
+			return;
+		}
+
+		ret = bt_le_ext_adv_set_data(adv_ext, ad_peer_local, ad_peer_local_size, NULL, 0);
+		if (ret) {
+			LOG_ERR("Failed to set advertising data. Err: %d", ret);
+			return;
+		}
+	}
+
+	ret = bt_le_ext_adv_start(adv_ext, BT_LE_EXT_ADV_START_DEFAULT);
+	if (ret) {
+		LOG_ERR("Failed to start advertising set. Err: %d", ret);
+		return;
+	}
+
+	LOG_INF("Advertising successfully started");
+}
+
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	int ret;
+	char addr[BT_ADDR_LE_STR_LEN];
+	uint16_t conn_handle;
+	enum ble_hci_vs_tx_power conn_tx_pwr;
+
+	if (err) {
+		return;
+	}
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	LOG_INF("Connected: %s", addr);
+
+	ret = bt_hci_get_conn_handle(conn, &conn_handle);
+	if (ret) {
+		LOG_ERR("Unable to get conn handle");
+	} else {
+#if (CONFIG_NRF_21540_ACTIVE)
+		conn_tx_pwr = CONFIG_NRF_21540_MAIN_DBM;
+#else
+		conn_tx_pwr = CONFIG_BLE_CONN_TX_POWER_DBM;
+#endif /* (CONFIG_NRF_21540_ACTIVE) */
+		ret = ble_hci_vsc_conn_tx_pwr_set(conn_handle, conn_tx_pwr);
+		if (ret) {
+			LOG_ERR("Failed to set TX power for conn");
+		} else {
+			LOG_DBG("TX power set to %d dBm for connection %p", conn_tx_pwr,
+				(void *)conn);
+		}
+	}
+
+	conn_set(bt_conn_ref(conn));
+}
+
+static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
+{
+	int ret;
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_INF("Disconnected: %s (reason 0x%02x)", addr, reason);
+
+	bt_conn_unref(conn);
+	conn_set(NULL);
+
+	ret = ble_mcp_conn_disconnected(conn);
+	if (ret) {
+		LOG_ERR("ble_msc_conn_disconnected failed with %d", ret);
+	}
+
+	k_work_submit(&adv_work);
+}
+
+static void security_changed_cb(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	int ret;
+
+	if (err) {
+		LOG_ERR("Security failed: level %d err %d", level, err);
+		ret = bt_conn_disconnect(conn, err);
+		if (ret) {
+			LOG_ERR("Failed to disconnect %d", ret);
+		}
+	} else {
+		LOG_DBG("Security changed: level %d", level);
+	}
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected_cb,
+	.disconnected = disconnected_cb,
+	.security_changed = security_changed_cb,
+};
+
+int bt_mgmt_ext_adv_start(const struct bt_data *ad_peer, size_t adv_size,
```

Comment:
```suggestion
int bt_mgmt_ext_adv_start(const struct bt_data *adv_peer, size_t adv_size,
```

Since we are using adv_size we should use adv_peer also?

Reply:
Should be changed all places if changed here

Reply:
Changed to ext_adv, ext_adv_size, per_adv, and per_adv_size for consistency